eng/scripts/Verify-And-Merge-PRs.ps1

```diff
@@ -22,43 +23,60 @@ if ($AuthToken) {
-foreach ($prDataLine in $PRData)
-{
-  $repoOwner, $repoName, $prNumber = $prDataLine.Split(";")
 
-  $prApiUrl = "https://api.github.com/repos/${repoOwner}/${repoName}/pulls/${prNumber}"
-  $prUrl = "https://github.com/${repoOwner}/${repoName}/pull/${prNumber}"
-  
-  try
+do 
+{
+  $unMergablePRs = @()
+  foreach ($prDataLine in $PRData)
   {
-    $response = Invoke-RestMethod -Headers $headers $prApiUrl
-    if ($response.merged) {
-      Write-Host "${prUrl} is merged."
-    }
-    elseif ($response.state -eq "closed") {
-      LogWarning "${prUrl} is closed. Please investigate why was not merged."
-      $ReadyForMerge = $false
-    }
-    elseif ($response.mergeable -and $response.mergeable_state -eq "clean") {
-      Write-Host "${prUrl} is ready to merge."
+    $repoOwner, $repoName, $prNumber = $prDataLine.Split(";")
 
-      $mergablePRs += @{ Url = $prApiUrl; HeadSHA = $response.head.sha }
-    }
-    elseif ($response.mergeable_state -ne "clean") {
-      LogWarning "${prUrl} is blocked ($($response.mergeable_state)). Please ensure all checks are green and reviewers have approved."
-      $ReadyForMerge = $false
+    $prApiUrl = "https://api.github.com/repos/${repoOwner}/${repoName}/pulls/${prNumber}"
+    $prUrl = "https://github.com/${repoOwner}/${repoName}/pull/${prNumber}"
+    
+    try
+    {
+      $response = Invoke-RestMethod -Headers $headers $prApiUrl
+      if ($response.merged) {
+        Write-Host "${prUrl} is merged."
+      }
+      elseif ($response.state -eq "closed") {
+        LogWarning "${prUrl} is closed. Please investigate why was not merged."
+        $ReadyForMerge = $false
+      }
+      elseif ($response.mergeable -and $response.mergeable_state -eq "clean") {
+        Write-Host "${prUrl} is ready to merge."
+
+        $mergablePRs += @{ Url = $prApiUrl; HeadSHA = $response.head.sha }
+      }
+      elseif ($response.mergeable_state -ne "clean") {
+        LogWarning "${prUrl} is blocked ($($response.mergeable_state)). Please ensure all checks are green and reviewers have approved."
+        $ReadyForMerge = $false
+        $unMergablePRs += $prDataLine
+      }
+      else {
+        LogWarning "${prUrl} is in an unknown state please contact engineering system team to understand the state."
+        LogWarning $response
+        $ReadyForMerge = $false
+        $unMergablePRs += $prDataLine
+      }
     }
-    else {
-      LogWarning "${prUrl} is in an unknown state please contact engineering system team to understand the state."
-      LogWarning $response
-      $ReadyForMerge = $false
+    catch {
+      LogError "Invoke-RestMethod ${prApiUrl} failed with exception:`n$_"
+      exit 1
     }
   }
-  catch {
-    LogError "Invoke-RestMethod ${prApiUrl} failed with exception:`n$_"
-    exit 1
+  if ($unMergablePRs.Count -gt 0)
+  {
+    $PRData = $unMergablePRs
+    Start-Sleep -Seconds 30
+    $RetryCount--
+    LogDebug "Retrying merging for unmergable Prs"
+    $ReadyForMerge = $true
```

Comment:
Should this be marked to true here? What happens if we still have unmergable stuff after the last retry and we break out of this loop and then we will go down and try to merge things.

Reply:
Fixed with this PR https://github.com/Azure/azure-sdk-tools/pull/1160